### DISCUSSION TITLE
[error] 1) sizeinit 2) beginning with -1 frame.

### DISCUSSION THREAD

**Discussion by wonwill** [YAMACS-SML/YAMACS#9] [Q&A]
I continuously get two messages.

**I. variable 'sizeinit' error**

The first one appeared as soon as I ran the mdrun script.
After stopping the gmx process and running again from the beginning, it works properly.
Could you help me out with this problem?
<img width="235" alt="CleanShot 2023-12-12 at 14 30 19@2x" src="https://github.com/YAMACS-SML/YAMACS/assets/5327542/f391793c-2042-43f1-92f3-bced023161fa">

**II. 'md_0_1.xtc' FileNotFound error and "-1" frame**

If I proceed as described above, the simulation begins with -1.
Is it correct? Or should any steps adjust the number of MD simulations?
<img width="787" alt="CleanShot 2023-12-12 at 15 42 19@2x" src="https://github.com/YAMACS-SML/YAMACS/assets/5327542/0bee73bf-e02d-4173-9eb9-625d0fa8c003">

**Comment by YAMACS-SML** [comment]
Hello, thank you for using YAMACS. Your feedback is always important for making YAMACS more useful. I suggest recording a short video of your entire simulation, from system preparation to MD-run, and uploading it here as a zip file. This will allow us to thoroughly analyze and debug any errors. From the screenshot, it is challenging to understand the mentioned error. Thank you for your cooperation.

**Comment by wonwill** [comment]
Sorry for my late reply and Happy New Year!
I hope to take advantage of the yamacs for my further research.

**I. variable 'sizeinit' error**

The error happened because different versions of gmx were installed due to different versions of ubuntu.
So I reinstalled new version of ubuntu and then install gmx again.
The yamacs works properly now.

**II. 'md_0_1.xtc' FileNotFound error and "-1" frame**

To pass this error, I always run the yamacs script again from the beginning.
The frame beginning from -1 happens almost always.

_The most important thing is how I can handle these outputs;_
i) Can I ignore the frame beginning from -1?
ii) What should I do to adjust these errors?
Because I generated many outputs in such a way.

Here are two examples including files and short videos.
If you need more files, please let me know.

amber03 error example
recording-yamacs_amber03_error_&minus-cycle_in_beginning.mp4
https://gofile.me/778YM/7KP0qnOHr
recording-yamacs_amber03_error&_minus-cycle_in_beginning.mp4.tar.gz
https://gofile.me/778YM/qxal9ggSy

charmm error example
recording-yamacs_charmm_error_&minus-cycle_in_beginning.mp4
https://gofile.me/778YM/IuYfN2wp0
recording-yamacs_charmm_error&_minus-cycle_in_beginning.tar.gz
https://gofile.me/778YM/vhCOuejkO

**Comment by YAMACS-SML** [comment]
Happy new Year. Thanks for detailing the encountered errors. Let's address each issue step by step:

Problem 1: 'md_0_1.xtc' File Not Found Error

The problem you faced is related to the usage of the `gmx mpi` tool, specifically in the context of parallel computation for creating the `.xtc` file and running the simulation using it. The parallel execution of both commands led to a file not found error during the MD run.

Resolution: We have updated the 3rd plugin (`YAMACS_3_MD.py`). Please replace the existing file with this updated version. In this version, we increased the time gap between generating the `.xtc` file and launching the simulation.

Problem 2: Beginning from -1 Frame

This issue arises due to an incomplete termination step. However, it's not necessary to perform the entire calculation again for the second attempt. In the first step, all the necessary calculations for the MD simulation were completed, but an error occurred during the 'md run' step. In the second attempt, you only need to run the third plugin. The lack of a plugin crash error during the second attempt is because the plugin found the existing `.xtc` file (incomplete) from the first run and used it for calculations.

Resolution: Remove the existing `md_0_1.xtc` file and run the updated 3rd plugin (`YAMACS_3_MD.py`).


I hope this refined explanation helps convey the information more clearly. If you have any specific questions or need further clarification, feel free to ask!

**Comment by wonwill** [comment]
Thanks for your immediate reply.

Problem 1 wasn't "completely" solved by the updated YAMACS_3_MD.py.

I guess that the error depends on the size of the target protein.
_If the protein size > 1 MB, "'md_0_1.xtc' File Not Found Error" happens again._
[4QEX_WT_monomer.pdb.zip](https://github.com/YAMACS-SML/YAMACS/files/13873328/4QEX_WT_monomer.pdb.zip)

I replaced all 'gmx' with 'gmx_mpi' in the YAMACS_3_MD script.
[YAMACS_3_MD.py.zip](https://github.com/YAMACS-SML/YAMACS/files/13873011/YAMACS_3_MD.py.zip)

In the previous version, this issue also happened while using gmx, not gmx_mpi.
Could you help me out with this issue again?

**Comment by wonwill** [comment]
Problem 2: Beginning from -1 Frame

The bad news is that I have realized recently that my students generated more than 60 outputs with this problem.
Only few outputs are labeled with either "Beginning from -1 Frame" or "Beginning at zero".
Therefore, I have to handle/correct all of them.
Do you help me out with this issue?

Furthermore, 
1) I got confused "the second attempt" which is either "after encountering File Not Found Error" or "MD simulation beginning from -1 Frame".
2) Should I remove all of them if different versions of xtc file exist, such as "md_0_1.xtc" and "#md_0_1.xtc.2#"?
I ran the updated 3_MD.py and the default mdp files after removing "md_0_1.xtc" only in the completed md simulation. But another error "protein.txt File Not Found Error" happened.

Here is one of my outputs beginning from -1.
https://gofile.me/778YM/HlpTYEXQw

**Comment by YAMACS-SML** [comment]
Hi, here is the calculation of your file that ran without any error. You will find the video inside it.  
https://gofile.io/d/fXqwf3
In this link you will get the video as well as the important  files.
